eclair-core/src/main/scala/fr/acinq/eclair/payment/receive/MultiPartHandler.scala

```diff
@@ -240,11 +244,14 @@ object MultiPartHandler {
     }
   }
 
-  private def validatePayment(payment: IncomingPacket.FinalPacket, record: IncomingPayment, currentBlockHeight: Long)(implicit log: LoggingAdapter): Option[CMD_FAIL_HTLC] = {
+  private def validatePayment(nodeParams: NodeParams, payment: IncomingPacket.FinalPacket, record: IncomingPayment)(implicit log: LoggingAdapter): Option[CMD_FAIL_HTLC] = {
     // We send the same error regardless of the failure to avoid probing attacks.
-    val cmdFail = CMD_FAIL_HTLC(payment.add.id, Right(IncorrectOrUnknownPaymentDetails(payment.payload.totalAmount, currentBlockHeight)), commit = true)
+    val cmdFail = CMD_FAIL_HTLC(payment.add.id, Right(IncorrectOrUnknownPaymentDetails(payment.payload.totalAmount, nodeParams.currentBlockHeight)), commit = true)
     val paymentAmountOk = record.paymentRequest.amount.forall(a => validatePaymentAmount(payment, a))
-    val paymentCltvOk = validatePaymentCltv(payment, record.paymentRequest.minFinalCltvExpiryDelta.getOrElse(Channel.MIN_CLTV_EXPIRY_DELTA).toCltvExpiry(currentBlockHeight))
+    // The default `min_final_cltv_expiry` we accept is our `fulfill_safety_before_timeout` to which we add one block for
+    // safety (to handle races between our peer acknowledging the HTLC fulfill and a new block being produced).
+    val defaultMinFinalCltvExpiryDelta = nodeParams.fulfillSafetyBeforeTimeoutBlocks + 1
```

Comment:
Shouldn't this logic take place in `validatePaymentCltv`? Why don't we rely on the value in `record.paymentRequest.minFinalCltvExpiryDelta`?

Reply:
Because unfortunately `record.paymentRequest.minFinalCltvExpiryDelta` is an `Option`...

Reply:
I can move that inside `validatePaymentCltv` though, that makes sense.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1483/commits/14c023b67b629338570d11815cc9cf9dd16c3216